pkg/certifier/scorecard/types.go

```diff
@@ -0,0 +1,31 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scorecard
+
+import (
+	"github.com/guacsec/guac/pkg/certifier"
+	sc "github.com/ossf/scorecard/v4/pkg"
+)
+
+// Scorecard is an interface for the scorecard library. This can also be mocked for testing.
+type Scorecard interface {
+	GetScore(repoName, commitSHA string) (*sc.ScorecardResult, error)
+}
+
+type CertQuerier interface {
```

Comment:
Yes, I agree that we keep them decoupled. Similar to what you said `QueryComponents` can be used by multiple certifiers in the future.

Reply:
The reason for creating `CertQuerier` interface was so that `scorecard` can act as a `certifier.Certifier` and as a `certifier.QueryComponents`. 

This is the reason for `CertQuerier`, similar to the golang standard library, composition of interfaces https://pkg.go.dev/io#ReadSeeker. 

How else do we solve this issue?

Reply:
I agree! I have removed `CertQuerier` and have moved `GetComponents` into a separate package.

https://github.com/guacsec/guac/pull/459/files#diff-3910d168bb8d504976061ce655af1cb672f93073885055b7feaf7cba2f14eef5